framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -234,6 +242,18 @@ class MaterialPropertyInterface
   /// Storage vector for MaterialProperty<Real> default objects
   std::vector<std::unique_ptr<MaterialProperty<Real>>> _default_real_properties;
 
+  /**
+   * Call this function to add the passed in material property ID to the list of material property
+   * IDs that _this_ object depends on.
+   *
+   * @param mat_prop_id The material property ID that should be added to _this_ object's material
+   * dependency list.
+   */
+  void addMatPropDependency(unsigned int mat_prop_id) { _material_property_dependencies.insert(mat_prop_id); }
```

Comment:
I don't think we need a protected function.  It is only used in this class and not meant to be overridden.  Better to just do the one line insert directly in each location and not have this function altogether.